acquia/cli
# CLI-745: Parallel unit tests

**Motivation**

Fixes #914

This reduces unit test suite time from 80 s to 20 s on my machine.

**Proposed changes**


- Run unit tests in parallel with paratest.
- Isolate test fixtures per-test to avoid race conditions.
- Remove extraneous code related to cache- and fixture-management now that tests run with isolated fixtures
- Forbid the use of sys_get_temp_dir, which is not thread-safe, and instead provide a safe $tmpDir via DI
- Split Coveralls reporting into a separate GitHub job since paratest behaves slightly differently with coverage enabled

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Run `composer test` and verify that tests run in reduced time
4. Check for regressions in common ACLI commands

**Merge requirements**
